Gemfile

```diff
@@ -29,6 +29,8 @@ gem 'puma', '~> 5.0'
 gem 'sass-rails', '>= 6'
 # An utility first CSS framework
 gem 'tailwindcss-rails', '~> 0.3.0'
+# A framework for building reusable, testable & encapsulated view components in Ruby on Rails
+gem 'view_component', '~> 2.24', require: 'view_component/engine'
```

Comment:
I'm not against `view_component` but I'm worried to continue to add dependencies that will make it harder for people to get started with the base.. thoughts?

Reply:
I don't like using more things than the default rails form helpers + CSS classes usually. But in this case with so many utility classes (thanks to Tailwind) could be a good way to make reusable components and make the UI more uniform

Reply:
I think this is more like "this is the way we do it", like a live playbook. I really don't like dependencies but in this case I think Tailwind was designed to be used with components, I don't see it as any other way. I've also thought of these while building this PR:

```ruby
# erb
<%= f.submit 'Sign in', class: button_css %>

# helper
module ButtonsHelper
  def button_css
    %w[....]
  end
end
```

And

```ruby
# erb
<%= button('Sign in') %>

# helper
module ButtonsHelper
  def button
    button_tag('Sign in', class: %w[....])
  end
end
```

I'm afraid the latter would collide with Rails' form helpers and that it's actually just components done in a functional manner

Reply:
> I really don't like dependencies but in this case I think Tailwind was designed to be used with components, I don't see it as any other way

@brunvez I think Tailwind would work just fine with plain Rails partials

Reply:
Yeah but how would you extract common css? I'm alright sprinkling. utility classes here and there, but when they are as many as the button have I'd rather keep them in one place away from the rest of the code

Reply:
you can have a shared button partial such as you created a button component, right?